File: uikit/src/main/java/viewmodel/ReceiptListViewModel.java
package viewmodel;

import android.content.Context;

import com.cometchat.pro.models.MessageReceipt;
import com.cometchat.pro.models.User;
import com.cometchat.pro.uikit.CometChatReceiptsList;
import com.cometchat.pro.uikit.CometChatUserList;

import java.util.List;

import adapter.ReceiptListAdapter;
import adapter.UserListAdapter;
import listeners.StickyHeaderDecoration;

public class ReceiptListViewModel {

    private static final String TAG = "ReceiptListViewModel";

    private  Context context;

    private ReceiptListAdapter receiptListAdapter;

    private CometChatReceiptsList receiptsList;



    public ReceiptListViewModel(Context context, CometChatReceiptsList receiptsList,
                                boolean showDelivery,boolean showRead){
        this.receiptsList = receiptsList;
        this.context=context;
        setReceiptListAdapter(receiptsList,showDelivery,showRead);
    }

    private ReceiptListViewModel(){

    }

    private ReceiptListAdapter getAdapter() {
        if (receiptListAdapter==null){
            receiptListAdapter=new ReceiptListAdapter(context);
        }
        return receiptListAdapter;
    }

    public void add(MessageReceipt messageReceipt){
        if (receiptListAdapter!=null)
            receiptListAdapter.add(messageReceipt);

    }
    public void add(int index,MessageReceipt messageReceipt){
        if (receiptListAdapter!=null)
            receiptListAdapter.addAtIndex(index,messageReceipt);

    }

    public void update(MessageReceipt messageReceipt){
        if (receiptListAdapter!=null)
            receiptListAdapter.updateReceipts(messageReceipt);

    }

    public void clear()
    {
        if (receiptListAdapter!=null)
            receiptListAdapter.clear();
    }
    private void setReceiptListAdapter(CometChatReceiptsList cometChatReceiptsList,
                                       boolean showDelivery,boolean showRead){
        receiptListAdapter=new ReceiptListAdapter(context);
        cometChatReceiptsList.showDelivery(showDelivery);
        cometChatReceiptsList.showRead(showRead);
        cometChatReceiptsList.setAdapter(receiptListAdapter);
    }

    public void setReceiptList(List<MessageReceipt> messageReceiptsList){
          getAdapter().updateList(messageReceiptsList);
    }

    public void update(int index, MessageReceipt messageReceipt) {
        if (receiptListAdapter!=null)
            receiptListAdapter.updateReceiptsAtIndex(index,messageReceipt);
    }
}

File: uikit-kotlin/src/main/java/viewmodel/ReceiptListViewModel.kt
package viewmodel

import adapter.ReceiptListAdapter
import android.content.Context
import com.cometchat.pro.models.MessageReceipt
import com.cometchat.pro.uikit.CometChatReceiptsList

class ReceiptListViewModel {
    private val TAG = "ReceiptListViewModel"

    private var context: Context? = null

    private var receiptListAdapter: ReceiptListAdapter? = null

    private var receiptsList: CometChatReceiptsList? = null

    constructor(context: Context?, receiptsList: CometChatReceiptsList?,
                showDelivery: Boolean, showRead: Boolean) {
        this.receiptsList = receiptsList
        this.context = context
        setReceiptListAdapter(receiptsList, showDelivery, showRead)
    }

    constructor()

    private fun getAdapter(): ReceiptListAdapter? {
        if (receiptListAdapter == null) {
            receiptListAdapter = ReceiptListAdapter(context)
        }
        return receiptListAdapter
    }

    fun add(messageReceipt: MessageReceipt?) {
        if (receiptListAdapter != null) receiptListAdapter!!.add(messageReceipt)
    }
    fun add(index: Int, messageReceipt: MessageReceipt?) {
        if (receiptListAdapter != null) receiptListAdapter!!.addAtIndex(index, messageReceipt)
    }
    fun update(messageReceipt: MessageReceipt?) {
        if (receiptListAdapter != null) receiptListAdapter!!.updateReceipts(messageReceipt)
    }
    fun clear() {
        if (receiptListAdapter != null) receiptListAdapter!!.clear()
    }

    private fun setReceiptListAdapter(cometChatReceiptsList: CometChatReceiptsList?,
                                      showDelivery: Boolean, showRead: Boolean) {
        receiptListAdapter = ReceiptListAdapter(context)
        cometChatReceiptsList!!.showDelivery(showDelivery)
        cometChatReceiptsList!!.showRead(showRead)
        cometChatReceiptsList!!.adapter = receiptListAdapter
    }

    fun setReceiptList(messageReceiptsList: List<MessageReceipt>) {
        getAdapter()!!.updateList(messageReceiptsList)
    }
    fun update(index: Int, messageReceipt: MessageReceipt?) {
        if (receiptListAdapter != null) receiptListAdapter!!.updateReceiptsAtIndex(index, messageReceipt)
    }
}